src/api/app/controllers/trigger_workflow_controller.rb

```diff
@@ -71,7 +71,10 @@ def create_workflow_run
     raise Trigger::Errors::InvalidToken, 'Wrong token type. Please use workflow tokens only.' unless @token.is_a?(Token::Workflow)
 
     request_headers = request.headers.to_h.keys.map { |k| "#{k}: #{request.headers[k]}" if k.match?(/^HTTP_/) }.compact.join("\n")
-    @workflow_run = @token.workflow_runs.create(request_headers: request_headers, request_payload: request.body.read)
+    # We need to write the payload into `request_json_payload` too, while the `request_payload` is still around
+    @workflow_run = @token.workflow_runs.create(request_headers: request_headers,
+                                                request_payload: request.body.read,
+                                                request_json_payload: request.body.read)
```

Comment:
When this PR is deployed, but before the migration adding `request_json_payload` is executed, there will be errors happening here. To prevent those errors, you'll have to split this PR. The first PR will have the migration. The second PR will have this code and the data migration to backfill data.